components/board/genericlinux/gpio.go

```diff
@@ -21,19 +21,15 @@ type gpioPin struct {
 	// the multiple calls to the gpio package don't have race conditions.
 	devicePath string
 	offset     uint32
+	line       *gpio.Line
```

Comment:
so when are lines closed now?

Reply:
When the program exits. In the future, they might be closed when you enable hardware PWMs, but that's not built yet.

Reply:
When you close a line, the GPIO outputs might no longer hold their state, hence why they should stay open for a long time

Reply:
what happens if the board gets reconfigured? should we not close those out or do they transfer over? want to just make sure we don't leak FDs.

Reply:
(and we didn't catch this last time because a third of the pins _do_ hold their state when you close the line, and it happens that I was testing on two of those and had assumed that they'd all behave identically, that if those two work then they all do.)

Reply:
> what happens if the board gets reconfigured?

That's a very good point, thanks! If the board struct gets destructed (or whatever it's called in Go), we should indeed close these. I'll see what I can do...

Reply:
sounds good. drive by complete 😎

Reply:
Excellent call, Eric! With the code from commit 55eb0ca, if you change your robot's config without restarting the server and then try setting the same GPIO pins over again, you get errors about the device already being in use (and even if we somehow got past that error, I imagine you're right that we're leaking file descriptors). 

I'll see what I can do to fix this...